.github/workflows/format.yml

```diff
@@ -1,26 +1,24 @@
 name: Scalafmt
 
+permissions: read-all
+
 on:
   pull_request:
     branches: ['**']
 
-env:
-  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-
 jobs:
   build:
     name: Code is formatted
-    strategy:
-      matrix:
-        os: [ubuntu-latest]
-    runs-on: ${{ matrix.os }}
+    runs-on: ubuntu-latest
     steps:
       - name: Checkout current branch (full)
         uses: actions/checkout@v2
         with:
           fetch-depth: 0
+          persist-credentials: false
 
       - name: Check project is formatted
-        uses: jrouly/scalafmt-native-action@v1
+        uses: jrouly/scalafmt-native-action@v2
         with:
-          version: '3.4.3'
+          version: '3.7.2'
+          arguments: '--list --mode diff-ref=origin/master'
```

Comment:
This change means that the scalafmt github action check will only run on files that have actually changed (according to `origin/master`) rather than checking that the entire codebase is formatting.